src/mpid/ch4/shm/xpmem/xpmem_init.c

```diff
@@ -50,12 +55,30 @@ int MPIDI_XPMEM_mpi_init_hook(int rank, int size, int *n_vcis_provided, int *tag
                                     (void **) &xpmem_segids, MPL_MEM_SHM);
     if (mpi_errno)
         MPIR_ERR_POP(mpi_errno);
+#ifdef MPIDI_CH4_SHM_XPMEM_COOP_P2P
+    mpi_errno = MPIDU_shm_seg_alloc(sizeof(uint64_t), (void **) &coop_caddr, MPL_MEM_SHM);
+    if (mpi_errno)
+        MPIR_ERR_POP(mpi_errno);
+#endif
     mpi_errno = MPIDU_shm_seg_commit(&shm_seg,
                                      &shm_seg_barrier,
                                      num_local, my_local_rank, local_rank_0, rank, MPL_MEM_SHM);
     if (mpi_errno)
         MPIR_ERR_POP(mpi_errno);
 
+#ifdef MPIDI_CH4_SHM_XPMEM_COOP_P2P
+    if (!my_local_rank) {
```

Comment:
Can you add a comment here explaining why rank `0` does not allocate any space for counters?

Reply:
Sure.
I also type the reason here:

> Only rank 0 need to actually allocate an array to store counters and other processes should attach to this array through MPIDI_XPMEM_seg_regist